app/services/doc_auth/socure/requests/document_request.rb

```diff
@@ -47,6 +47,47 @@ def handle_http_response(http_response)
           JSON.parse(http_response.body, symbolize_names: true)
         end
 
+        def handle_invalid_response(http_response)
+          if http_response.nil? || !http_response.body.present?
+            status = nil
+            status = http_response.status unless http_response.nil?
+            message = [self.class.name, 'Unexpected HTTP response', status].join(' ')
+            exception = DocAuth::RequestError.new(message, status)
+            return handle_connection_error(exception: exception)
+          end
+          message = [
+            self.class.name,
+            'Unexpected HTTP response',
+            http_response.status,
+          ].join(' ')
+          exception = DocAuth::RequestError.new(message, http_response.status)
+
+          response_body = begin
+            JSON.parse(http_response.body)
```

Comment:
```suggestion
            JSON.parse(http_response.body)
```

will revert as per chat to check for nil